### PR TITLE
[TSK-205] added and configured kafka to notification and payment manager

### PR DESCRIPTION
**Pull request by Hoocki**

### CODE REVIEW

File: notification-manager/src/main/resources/application.yaml [hunk 1]
```diff
@@ -15,8 +15,23 @@ spring:
           starttls:
             enable: true
           auth: true
+  kafka:
+    bootstrap-servers: localhost:9092
```
Comment (adsukharev) [inline notification-manager/src/main/resources/application.yaml]: localhost is for local

File: notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/EmailNotificationService.java [hunk 2]
```diff
@@ -18,6 +20,7 @@ public class EmailNotificationService implements NotificationService {
     public void sendMessage(final PaymentNotification paymentNotification) {
         final var message = mailMessageGenerator.getPaymentMessage(paymentNotification);
         mailSender.send(message);
+        log.info("Message sent to: {}", paymentNotification.emailTo());
```
Comment (adsukharev) [inline notification-manager/src/main/java/com/qpay/notificationmanager/service/impl/EmailNotificationService.java:23]: You have 2 `Message sent` logs but meaning is different. In one case it is an email, in other - event

File: payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentProducer.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+public class PaymentProducer {
+
+    private final KafkaTemplate<String, PaymentNotification> kafkaTemplate;
+
+    @Value("${kafka.topics.payment-topic}")
+    private String paymentTopic;
+
+    public void sendPayment(final PaymentNotification paymentNotification) {
+        kafkaTemplate.send(paymentTopic, paymentNotification.emailTo(), paymentNotification);
+        log.info("Message send {}", paymentNotification);
```
Comment (adsukharev) [inline payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentProducer.java]: sent

File: payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/PaymentServiceImpl.java [hunk 4]
```diff
@@ -43,7 +41,13 @@ public WalletEntity makePayment(final WalletPayment walletPayment) {
         final var updatedToWalletBalance = toWallet.getBalance().add(walletPayment.amount());
         updateBalance(updatedToWalletBalance, toWallet);
 
-        executorService.execute(() -> notificationClient.sendNotification(walletPayment));
+        final var notificationCreation = PaymentNotification.builder()
```
Comment (adsukharev) [inline payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/PaymentServiceImpl.java]:
```suggestion
        final var paymentNotification = PaymentNotification.builder()
```

File: payment-manager/src/main/resources/application.yaml [hunk 5]
```diff
@@ -5,6 +5,10 @@ webclient:
     transaction-history-manager:
       base-url: http://localhost:8084
 
+kafka:
```
Comment (adsukharev) [inline payment-manager/src/main/resources/application.yaml]: you have 2 kafka keys in the same config. It is confusing. Find another name. Same for consumer

File: payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentProducer.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+@Component
+@RequiredArgsConstructor
+public class PaymentProducer {
+
+    private final KafkaTemplate<String, PaymentNotification> kafkaTemplate;
+
+    @Value("${kafka.topics.payment-topic}")
+    private String paymentTopic;
+
+    public void sendPayment(final PaymentNotification paymentNotification) {
```
Comment (adsukharev) [inline payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentProducer.java]:
```suggestion
    public void publishPaymentNotification(final PaymentNotification paymentNotification) {
```

File: payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentProducer.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class PaymentProducer {
```
Comment (adsukharev) [inline payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentProducer.java]:
```suggestion
public class PaymentNotificationProducer {
```

File: notification-manager/src/main/java/com/qpay/notificationmanager/event/PaymentConsumer.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import com.qpay.notificationmanager.service.NotificationService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class PaymentConsumer {
```
Comment (adsukharev) [inline notification-manager/src/main/java/com/qpay/notificationmanager/event/PaymentConsumer.java]:
```suggestion
public class PaymentNotificationConsumer {
```

File: payment-manager/src/main/resources/application.yaml [hunk 9]
```diff
@@ -5,6 +5,10 @@ webclient:
     transaction-history-manager:
       base-url: http://localhost:8084
```
Comment (adsukharev) [inline payment-manager/src/main/resources/application.yaml]: These URLs are for local

File: notification-manager/src/main/resources/application-DEV.yaml [hunk 10]
```diff
@@ -1,9 +1,13 @@
+spring:
+  kafka:
+    bootstrap-servers: localhost:9092
```
Comment (adsukharev) [inline notification-manager/src/main/resources/application-DEV.yaml:3]: same for payment manager

File: payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/PaymentServiceImpl.java [hunk 11]
```diff
@@ -41,12 +41,12 @@ public WalletEntity makePayment(final WalletPayment walletPayment) {
...
         updateBalance(updatedToWalletBalance, toWallet);
 
-        final var notificationCreation = PaymentNotification.builder()
+        final var paymentNotification  = PaymentNotification.builder()
                 .amount(walletPayment.amount())
                 .emailFrom(walletPayment.emailFrom())
                 .emailTo(walletPayment.emailTo())
                 .build();
-        paymentProducer.sendPayment(notificationCreation);
+        paymentNotificationProducer.publishPaymentNotification(paymentNotification );
```
Comment (adsukharev) [inline payment-manager/src/main/java/com/qpay/paymentmanager/service/impl/PaymentServiceImpl.java]: too many empty lines in this method. Probably it means this method has a lot of code or different logic and it is hard to read. Most likely this is a reason why you use empty lines. Either remove it or restructure code by extracting code to new methods.

File: payment-manager/src/test/java/com/qpay/paymentmanager/service/impl/PaymentServiceImplTest.java [hunk 12]
```diff
@@ -97,7 +97,7 @@ void should_makePaymentBetweenWallets() {
         given(walletService.getWalletById(toId)).willReturn(toWallet);
         given(walletRepository.save(expectedFromWallet)).willReturn(expectedFromWallet);
         given(walletRepository.save(expectedToWallet)).willReturn(expectedToWallet);
-        willDoNothing().given(executorService).execute(any());
+        willDoNothing().given(paymentNotificationProducer).publishPaymentNotification(any());
```
Comment (adsukharev) [inline payment-manager/src/test/java/com/qpay/paymentmanager/service/impl/PaymentServiceImplTest.java]: In general it's a bad practice to use `any()`. You should validate that exact data passed to this method is correct.

File: payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentNotificationProducer.java [hunk 13]
```diff
@@ -10,16 +10,16 @@
...
 
-    @Value("${kafka.topics.payment-topic}")
+    @Value("${broker.topics.payment-topic}")
     private String paymentTopic;
 
-    public void sendPayment(final PaymentNotification paymentNotification) {
+    public void publishPaymentNotification(final PaymentNotification paymentNotification) {
         kafkaTemplate.send(paymentTopic, paymentNotification.emailTo(), paymentNotification);
-        log.info("Message send {}", paymentNotification);
+        log.info("Event of sending message to kafka: {}", paymentNotification);
```
Comment (adsukharev) [inline payment-manager/src/main/java/com/qpay/paymentmanager/event/PaymentNotificationProducer.java]: Strange log from english point of view. Maybe just `Payment notification event sent {}`

File: notification-manager/src/main/resources/application-DEV.yaml (hunk 10 above)
Comment (Hoocki) [reply notification-manager/src/main/resources/application-DEV.yaml:3]: i don't get it what should i do. How i understand, move every single port that based on localhost to DEV.yml, am i correct?

Comment (adsukharev) [reply notification-manager/src/main/resources/application-DEV.yaml:3]: correct. Check payment-manager application.yaml. There's bootstrap-servers: localhost:9092 which should be moved as well

**Review by adsukharev — APPROVED** [review]
Don't merge until Kostya review all your PRS. If he is slow, push him. You have review his PRs as well

**Review by kostya4201337 — APPROVED** [review]

**Pull request merged**